File: frontend/src/types/group.ts
export interface Group {
  id: string;
  name: string;
  description?: string;
  type: 'project' | 'department' | 'company' | 'custom';
  creatorId: string;
  creatorName: string;
  status: 'active' | 'archived' | 'deleted';
  visibility: 'public' | 'private' | 'department';
  avatar?: string;
  maxMembers: number;
  currentMemberCount: number;
  allowFileSharing: boolean;
  allowMemberInvite: boolean;
  createdAt: string;
  updatedAt?: string;
  members: GroupMember[];
  isUserMember: boolean;
  userRole?: string;
}

export interface GroupMember {
  id: string;
  employeeId: string;
  employeeName: string;
  employeeEmail: string;
  employeePosition: string;
  employeeDepartment: string;
  employeeAvatar?: string;
  role: 'admin' | 'moderator' | 'member';
  status: 'active' | 'inactive' | 'banned';
  joinedAt: string;
}

export interface CreateGroupRequest {
  name: string;
  description?: string;
  type: 'project' | 'department' | 'company' | 'custom';
  visibility: 'public' | 'private' | 'department';
  maxMembers: number;
  allowFileSharing: boolean;
  allowMemberInvite: boolean;
  initialMembers: string[];
}

export interface UpdateGroupRequest {
  name?: string;
  description?: string;
  visibility?: 'public' | 'private' | 'department';
  maxMembers?: number;
  allowFileSharing?: boolean;
  allowMemberInvite?: boolean;
}

export interface AddGroupMemberRequest {
  employeeId: string;
  role: 'admin' | 'moderator' | 'member';
}

export interface UpdateGroupMemberRequest {
  role: 'admin' | 'moderator' | 'member';
  status: 'active' | 'inactive' | 'banned';
}

export interface GroupStatistics {
  totalMembers: number;
  totalFiles: number;
  createdAt: string;
  lastActivity: string;
}

File: backend/StaffManagementSystem/Models/Group.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace StaffManagementSystem.Models
{
    public class Group
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [Required(ErrorMessage = "Group name is required")]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        [StringLength(500)]
        public string? Description { get; set; }

        [Required]
        [StringLength(20)]
        public string Type { get; set; } = "project"; // department, project, company, custom

        [Required]
        [ForeignKey("Creator")]
        public string CreatorId { get; set; } = string.Empty;

        [StringLength(20)]
        public string Status { get; set; } = "active"; // active, archived, deleted

        [StringLength(20)]
        public string Visibility { get; set; } = "private"; // public, private, department

        [StringLength(500)]
        public string? Avatar { get; set; }

        public int MaxMembers { get; set; } = 100;

        public bool AllowFileSharing { get; set; } = true;

        public bool AllowMemberInvite { get; set; } = false; // Only admins can invite by default

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        // Navigation properties
        public User? Creator { get; set; }
        public ICollection<GroupMember> Members { get; set; } = new List<GroupMember>();
        public ICollection<SharedFile> SharedFiles { get; set; } = new List<SharedFile>();

        public Group()
        {
            Id = Guid.NewGuid().ToString();
        }
    }
}